### PR TITLE
Gadgets & Gizmos

### PR DESCRIPTION
**Pull request by Mechoid**
Adds Barbed Wire. Slows and damages individuals trying to cross it. Can be electrified if placed on a wire node. Can go dull from damage, or continual use.

Adds Sandbags. Climbable like tables, will allow adjacent individuals to fire over unhindered, while affording protection from projectiles crossing them.

Adds a modified power cell to the antag uplink. Electrifies walls and reinforced flooring adjacent to it. Always active.

Barricades are no longer in MACHINES? WHY.

### CODE REVIEW

**Comment by Novacat** [comment]
We World War One now

**Comment by Atermonera** [comment]
Please provide evidence of your tests for this.

**Comment by Mechoid** [comment]
image
image

File: code/game/objects/items/weapons/traps.dm [hunk 1]
```diff
@@ -158,3 +164,238 @@
...
+			"You hear the rustling of [material.name]."
+			)
+
+		if (do_after(user, 60))
+			user.visible_message(
+				"<span class='danger'>[user] has deployed \the [src].</span>",
+				"<span class='danger'>You have deployed \the [src]!</span>",
+				"You hear the rustling of [material.name]."
+				)
+			playsound(src, 'sound/machines/click.ogg',70, 1)
```
Comment (Atermonera) [inline code/game/objects/items/weapons/traps.dm:233]: Have you considered using wirecutter snip sound here?

File: code/datums/wires/mines.dm [hunk 2]
```diff
@@ -48,6 +55,15 @@
...
 			spawn(20)
 				C.explode()
+
+		if(WIRE_TRAP)
+			C.visible_message("[bicon(C)] *click!*", "[bicon(C)] *click!*")
+
+			if(mended)
+				C.visible_message("[bicon(C)] - The mine recalibrates[C.camo_net ? ", revealing \the [C.trap] inside." : "."]")
+
+				C.alpha = 255
```
[the remark this answers is not in the document]
Comment (Mechoid) [reply code/datums/wires/mines.dm:54]: If you r-click a turf, it does show up. Alpha is used since it's the same system changelings and ninjas use atm.

**Pull request merged**